rustls/src/client/hs.rs

```diff
@@ -625,7 +632,7 @@ impl State for ExpectServerHello {
         server_hello
             .random
             .write_slice(&mut self.randoms.server);
-        self.handshake.session_id = server_hello.session_id;
+        self.session_id = server_hello.session_id;
```

Comment:
NIT: IMO this would have been better as `let session_id = `; there's no need to modify `self` in this function and the eventual goal is to make all these functions as close to purely functional as possible

Reply:
Fair enough; we'll pick it up in #596.